src/modules/ModuleRunner.ts

```diff
@@ -72,7 +183,14 @@ export class ModuleRunner {
      * @param factory The module factory.
      */
     public registerModule(factory: ModuleFactory): void {
-        this.modules.push(new AppModule(factory));
+        const appModule = new AppModule(factory);
+
+        this.modules.push(appModule);
+
+        /**
+         * Check if the new module provides any extensions, and also ensure a given extension is only provided by a single runtime module
+         */
```

Comment:
no need for a doc comment here.

```suggestion
        // Check if the new module provides any extensions, and also ensure a given extension is only provided by a single runtime module.
```